Common/src/gRPC/Services/IAgent.cs

```diff
@@ -97,13 +99,16 @@ public interface IAgent : IDisposable
   /// <summary>
   ///   Submit tasks with payload already existing
   /// </summary>
-  /// <param name="request">Requests containing the tasks to submit</param>
+  /// <param name="requests">Requests containing the tasks to submit</param>
   /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
   /// <returns>
-  ///   Reply sent to the worker with the submitted tasks
+  ///   Submitted tasks to send to the worker
   /// </returns>
-  Task<SubmitTasksResponse> SubmitTasks(SubmitTasksRequest request,
-                                        CancellationToken  cancellationToken);
+  Task<ICollection<TaskCreationRequest>> SubmitTasks(ICollection<TaskSubmissionRequest> requests,
```

Comment:
I think it is a bit weird to return a TaskCreationRequest

Reply:
But that's what we need.

Reply:
Maybe the name is not right.